Rt685 update clock init

EmilioCBen opened the pull request:
Updated the clock_init file in the SOC of the RT6xx
to align with the latest SDK.
Added support for safe flash clock initialization in the RT6xx.

mmahadevan108 commented on boards/arm/mimxrt685_evk/Kconfig.board:
```diff
@@ -4,5 +4,6 @@
 config BOARD_MIMXRT685_EVK
 	bool "NXP MIMXRT685-EVK"
 	depends on SOC_SERIES_IMX_RT6XX
+	select CODE_DATA_RELOCATION_SRAM
```
This should be done in the `soc` as the file that is relocated into RAM is located in the `soc` folder

EmilioCBen replied:
@mmahadevan108 
I have moved this selection into the soc/Kconfig.series, hopefully this is a better spot.

DerekSnell reviewed (changes requested):
Thanks @EmilioCBen ,
Overall this is looking good.  But there are some changes I think we should make.

Thanks

EmilioCBen committed:
soc: Add safe flash clock support for MXRT685

Added Safe flash clock setup files for rt6xx

Signed-off-by: Emilio Benavente <emilio.benavente@nxp.com>

EmilioCBen committed:
soc: Setup up Kconfigs for safe flash code to load in SRAM

Added a selecion on rt6xx that allows the loading of
the safe flash clock initialization to be inside SRAM
if the user code is being executed from flash.

Signed-off-by: Emilio Benavente <emilio.benavente@nxp.com>

EmilioCBen committed:
soc: Updated clock_init in rt6xx

Updated the clock_init function to the latest sdk
and added a safe initialization for the flash setup

Signed-off-by: Emilio Benavente <emilio.benavente@nxp.com>

DerekSnell reviewed (dismissed):
Thanks @EmilioCBen ,
Looks good to me

dleach02 commented on soc/arm/nxp_imx/rt6xx/Kconfig.defconfig.series:
```diff
@@ -58,4 +58,8 @@ endif # MBEDTLS
 
 source "soc/arm/nxp_imx/rt6xx/Kconfig.defconfig.mimxrt6*"
 
+choice CODE_LOCATION
```
Move this to the board defconfig

EmilioCBen replied:
Thank you, I have now moved this choice into the board section.

mmahadevan108 commented on soc/arm/nxp_imx/rt6xx/soc.c:
```diff
@@ -24,12 +24,15 @@
 #include <fsl_clock.h>
 #include <fsl_common.h>
 #include <fsl_device_registers.h>
+#include "flash_clock_setup.h"
```
Would this need to be wrapped inside
`#ifdef CONFIG_CODE_FLEXSPI`

EmilioCBen replied:
Sure, I can do that.

danieldegrasse commented on soc/arm/nxp_imx/rt6xx/Kconfig.series:
```diff
@@ -10,5 +10,6 @@ config SOC_SERIES_IMX_RT6XX
 	select CPU_CORTEX_M_HAS_DWT
 	select SOC_FAMILY_IMX
 	select CLOCK_CONTROL
+	select CODE_DATA_RELOCATION_SRAM
```
Since we only relocate files to SRAM when `CONFIG_CODE_FLEXSPI=y`, this needs to be set to `select CODE_DATA_RELOCATION_SRAM if CODE_FLEXSPI`

EmilioCBen replied:
Thank you, I have added this change.
